mtags-shared/src/main/scala/scala/meta/internal/pc/InlayHints.scala

```diff
@@ -123,6 +186,36 @@ object InlayHints {
   }
 }
 
+final case class InlayHintBlock(
+    indentLevel: Int,
+    hints: List[BlockInlayHint]
+) {
+  def build: List[BlockInlayHint] = {
+    if (hints.length == 1) {
+      val head = hints.head
+      if (head.pos.getStart.getLine == head.pos.getEnd.getLine) Nil
+      else hints
+    } else
+      hints.map { hint =>
+        val naiveIndent = indentLevel - hint.pos.getEnd.getCharacter
+        val labels =
+          if (naiveIndent <= 0) hint.labels
+          else LabelPart(" ".repeat(naiveIndent)) :: hint.labels
+        BlockInlayHint(
```

Comment:
We could use hint.copy(labels = labels) here - for example if someone adds new fields to BlockInlayHint, copy() will handle them automatically